pkg/gateway/model_build_targetgroup.go

```diff
@@ -320,20 +335,16 @@ func (t *latticeServiceModelBuildTask) buildHTTPTargetGroupSpec(ctx context.Cont
 		svc := &corev1.Service{}
 		if err := t.Client.Get(ctx, serviceNamespaceName, svc); err != nil {
 			glog.V(6).Infof("Error finding backend service %v error :%v \n", serviceNamespaceName, err)
-			return latticemodel.TargetGroupSpec{}, err
+			if !isDeleted {
+				//Return error for creation request only,
+				//For k8sService deletion request, we should go ahead to build TargetGroupSpec model
+				return latticemodel.TargetGroupSpec{}, err
```

Comment:
return error for creation request only,
For k8sService deletion request, we should go ahead to build TargetGroupSpec model